### ISSUE TITLE
False positive in missing-template-string-indicator

### ISSUE THREAD

**Issue by Sjord** [returntocorp/sgrep-rules#1461]
[missing-template-string-indicator](https://github.com/returntocorp/semgrep-rules/blob/develop/javascript/lang/correctness/missing-template-string-indicator.yaml) looks for `{...}` between backticks. But it doesn't agree with JavaScript on what the start and end backticks are.

```
`test`;
if (true) { a = 3; }
`test`;
```

```
$ semgrep --config 'r/javascript.lang.correctness.missing-template-string-indicator'
using config from https://semgrep.dev/r/javascript.lang.correctness.missing-template-string-indicator. Visit https://semgrep.dev/registry to see all public rules.
downloading config...
running 1 rules...
corpus.js
severity:error rule:javascript.lang.correctness.missing-template-string-indicator.missing-template-string-indicator: This looks like a JavaScript template string. Are you missing a '$' in front of '{...}'?
1:`test`;
2:if (true) { a = 3; }
3:`test`;
ran 1 rules on 1 files: 1 findings
```

It found `{ a = 3; }` between backticks, but in this case that's not part of the template string.

**Comment by Sjord** [comment]
Ping @minusworld, @mschwager.

**Comment by inkz** [comment]
working on it https://github.com/returntocorp/semgrep-rules/pull/1473 :+1:
